### PR TITLE
Add rich text to itsm and project's tasks (and their templates)

### PR DESCRIPTION
**Pull request by orthagh**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

Internal ref 19546

### CODE REVIEW

File: inc/define.php [hunk 1]
```diff
@@ -498,7 +498,7 @@
          'marketplace' => ['marketplace']
       ]
    ],
-   'admin'        => ['clipboard'],
+   'admin'        => ['clipboard', 'tinymce'],
```
Comment (cedric-anne) [inline inc/define.php:502]: Why did you add it in admin section ?

**Comment by cedric-anne** [comment]
Maybe a migration is needed for this. Indeed, existing templates description uses `\n` char for new lines (which was correctly rendered on textarea field), but these `\n` are not converted into `<br/>` when editing them, so everything is displayed into a single line in WYSIWYG editor.

**Comment by orthagh** [comment]
fixed

File: inc/projecttask.class.php [hunk 2]
```diff
@@ -614,7 +614,10 @@ function projecttasktemplate_update(value) {
                $("#textfield_name'.$rand_name.'").val(data.name);
 
                // set textarea description
-               $("#description'.$rand_description.'").val(data.description);
+               $("#description'.$rand_description.'").html(data.description);
+               if (tasktinymce = tinymce.get("description'.$rand_description.'")) {
+                  tasktinymce.setContent(data.description.replace(/\r?\n/g, "<br />"));
+               }
```
Comment (cedric-anne) [inline inc/projecttask.class.php:620]: 1. Unless I miss something, you set textarea contents by redefining html content of textarea and by the setContent method of tinyMce. Only one should be used.

2. As task template is defined from a rich text, replacement of \n to <br> seems useless.

Comment (orthagh) [reply inc/projecttask.class.php:620]: it's copy pasta from another parts of the code. Thanks for raising this btw.
I'll try to fix both parts in the morning

Comment (orthagh) [reply inc/projecttask.class.php:620]: for point 1, i think it was because before, we had options to set glpi in simple or rich text.
Not the case anymore. So we can drop some code
